docs/design/clustering.md

```diff
@@ -0,0 +1,56 @@
+# Clustering in Kubernetes
+
+
+## Overview
+The term "clustering" refers to the process of having all members of the kubernetes cluster find and trust each other.  There are multiple different ways to achieve clustering with different security and usability profiles.  This document attempts to lay out the user experiences for clustering that Kubernetes aims to address.
+
+Once a cluster is established, the following is true:
+
+1. **Master -> Node**  The master needs to know which nodes can take work and what their current status is wrt capacity.
+  1. **Location** The master knows the name and location of all of the nodes in the cluster.
+  2. **Target AuthN** A way to securely talk to the kubelet on that node.  Currently we call out to the kubelet over HTTP.  This should be over HTTPS and the master should know what CA to trust for that node.
+  3. **Caller AuthN/Z** Currently, this is only used to collect statistics as authorization isn't critical.  This may change in the future though.
```

Comment:
This is saying that the kubelet needs to validate that the person who is talking to it's API is authn/z'd to talk to it's API.